plugins/org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/refactoring/NewFeatureNameUtil.java

```diff
@@ -121,10 +121,18 @@ protected String getFancyDefaultName(XExpression expression) {
 		if(expression instanceof XClosure)
 			return "function";
 		LightweightTypeReference type = batchTypeResolver.resolveTypes(expression).getActualType(expression);
-		if (type != null) {
-			if (type.isPrimitive())
+		return toVariableName(type);
+	}
+	
+	protected String toVariableName(LightweightTypeReference type) {
+		if(type != null) {
+			if(type.isPrimitive())
 				return type.getSimpleName().substring(0, 1);
-			else
+			else if(type.isArray()) 
+				return toVariableName(type.getComponentType()) + 's';
```

Comment:
Yes, the API may not be optimal for this use case. On the other hand, the current Java project carries some user settings how variables are supposed to be named and we could (should) use these were possible to provide a consistent experience. Maybe you can factor the important parts into something easier to reuse?